File: tests/src/test/java/com/optivem/atddaccelerator/templategenerator/systemtest/util/ProcessResult.java
package com.optivem.atddaccelerator.templategenerator.systemtest.util;

import lombok.Data;

@Data
public class ProcessResult {
    private final int exitCode;
    private final String output;
    private final String errors;

    public boolean isSuccess() {
        return exitCode == 0;
    }

    public boolean isError() {
        return !isSuccess();
    }
}

File: console/tests/Util/Process/ProcessResult.cs
namespace Optivem.AtddAccelerator.TemplateGenerator.SystemTests.Util.Process
{
    public class ProcessResult
    {
        public int ExitCode { get; }
        public string Output { get; }
        public string Errors { get; }

        public ProcessResult(int exitCode, string output, string errors)
        {
            ExitCode = exitCode;
            Output = output;
            Errors = errors;
        }

        public bool IsSuccess => ExitCode == 0;
        public bool IsError => !IsSuccess;
    }
}